components/sponsored_benefits/app/controllers/sponsored_benefits/organizations/plan_design_proposals/plans_controller.rb

```diff
@@ -28,6 +28,7 @@ def dental_reference_plans
         offering_query = ::Queries::EmployerPlanOfferings.new(plan_design_organization)
         @plans = offering_query.dental_reference_plans_by_id(params[:plans_ids], params[:active_year])
         @plans = @plans.select{|a| a.premium_tables.present?}
+        @plans_ids_json = @plans&.map{|a| a.id.to_s}&.to_json
```

Comment:
you've already have access to `params[:plan_ids]` here; 

`@plan_ids_json = params[:plans_ids].to_json` do the same, so you don't have to loop through.

Reply:
Because the vulnerability was around directly putting what they gave us into the view, this is essentially just confirming that the ids they gave were valid IDs for existing plans. In this case we'll either get the same valid IDs they gave, or it will display nothing if they gave false IDs.